elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreResourceDefinition.java

```diff
@@ -291,6 +294,16 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
             String relativeTo = asStringIfDefined(context, RELATIVE_TO, model);
             ServiceTarget serviceTarget = context.getServiceTarget();
 
+            if (type == null || type.equals(KeyStoreCredentialStore.KEY_STORE_CREDENTIAL_STORE)) {
+                implementationAttributes.putIfAbsent(CS_KEY_STORE_TYPE_ATTRIBUTE, "JCEKS");
+            }
+
+            List<String> filebasedKeystoreTypes = Arrays.asList("JKS", "JCEKS", "PKCS12");
+            String implAttrKeyStoreType = implementationAttributes.get(CS_KEY_STORE_TYPE_ATTRIBUTE);
+            if (location == null && implAttrKeyStoreType != null && filebasedKeystoreTypes.stream().anyMatch(implAttrKeyStoreType::equalsIgnoreCase)) {
```

Comment:
We probably should avoid a strem / lambda here - we should just be able to convert the value to upper case and compare against the collection.